scripts/download.sh

```diff
@@ -34,6 +34,22 @@ else
   echo "Snapshot already present"
 fi
 
-if [ ! -z "$STATS_IP" ]; then
-  echo -n "ctlstore.reflector.init_snapshot_download_time:$(($END - $START))|h|#$TAGS" | nc -u -w1 $NODE_IP $STATS_PORT
-fi
\ No newline at end of file
+emit_metrics(){
+  if [ ! -z "$STATS_IP" ]; then
+    counter=0
+    while ! echo exit | nc -u -w1 $NODE_IP $STATS_PORT;
+    if (($((counter % 15)) == 0)); then
+      echo "awaiting datadog UDP port to be ready..."
+    fi
+    counter=$((counter+1))
+    if ((counter > 300)); then
+      break;
+    fi
+    do sleep 1;
+    done
+
+    echo -n "ctlstore.reflector.init_snapshot_download_time:$(($END - $START))|h|#$TAGS" | nc -u -w1 $NODE_IP $STATS_PORT
+  fi
+}
+
+emit_metrics &
```

Comment:
I am trying to assign emit_metrics to a separate process so that it won't block reflector, would this work?

Reply:
I don't think so. The InitContainer itself is blocking the reflector container from coming up. This will either cause the script to return right away (which would defeat the whole point) or still block. 

Right now, its about 1 minute to download the prod ldb. Polling for 5 minutes in the worst case puts us at 6 minutes, right where we are now. 

The important distinction is that during the 5 minutes waiting to emit the metric, data plane apps are still able to start up and make progress. The data will potentially just be stale.

Reply:
Got it, makes a lot more sense!